jax/experimental/compilation_cache/file_system_cache.py

```diff
@@ -37,6 +39,36 @@ def put(self, key: str, value: bytes):
     """Adds new cache entry, possibly evicting older entries."""
     if not key:
       raise ValueError("key cannot be empty")
-      #TODO(colemanliyah):implement eviction
-    with open(os.path.join(self._path, key), "wb") as file:
+    if self._evict_entries_if_necessary(key, value):
+      self._create_a_file(self._path, key, value)
+    else:
+      warnings.warn("This compilation file is larger than the size of the cache")
```

Comment:
It's nice to include `value`'s size and the max size in the warning, so people can understand exactly what's going wrong. Users may not understand what "compilation file" means either. I suggest something like:
`f"Cache value of size {len(value)} is larger than the max cache size of {self._max_cache_size_bytes}"`
(If you haven't seen it before, the above is a [formatted string literal](https://docs.python.org/3/tutorial/inputoutput.html#formatted-string-literals). It's the best way to include variable values in a string in Python.)